### ISSUE TITLE
Export Localization failed because of "No such module 'CrashReporter'" in DatadogSDKCrashReporting

### ISSUE THREAD

**Issue by khsora34** [DataDog/dd-sdk-ios#1260]
### The issue

📝 Give us the error message you receive, describe the problem and answer the questions.

I tried to use the export localization option built in Xcode (Product -> Export Localizations) to export any localizable string in our files.

However, this option always fails to build with the message `No such module 'CrashReporter'` in the `DatadogSDKCrashReporting` pod.

<img width="481" alt="Screenshot 2023-04-19 at 10 28 15 AM" src="https://user-images.githubusercontent.com/32332953/233393678-6d77a028-abb2-4418-b200-3475d34e791b.png">

I also have an example project reproducing this issue. Maybe I'm missing some configuration steps?

[LocalizationTest.zip](https://github.com/DataDog/dd-sdk-ios/files/11287183/LocalizationTest.zip)

---

#### Datadog SDK version:

_Which version of the Datadog SDK causes this problem? e.g. `1.2.0`_

`1.17.0`

#### Last working Datadog SDK version:

_What is the last Datadog SDK version where this problem didn't occur? e.g. `1.1.0`_

**N/A**
We started with this version.

#### Dependency Manager:

_Which dependency manager do you use? e.g. Cocoapods / Carthage / SPM / ..._

**Cocoapods**

#### Other toolset:

_Do you use additional tools with your dependency manager? e.g. [CarthageCache](https://github.com/Wolox/carthage_cache)_

**No**

#### Xcode version:

_e.g. `Xcode 11.5 (11E608c)`_

`Xcode 14.3 (14E222b)`

#### Swift version:

_e.g. `5.1`_

`5.8`

#### Deployment Target:

_What is the Deployment Target of your app? e.g. `iOS 12`, `iPhone` + `iPad`_

`iOS 16.2`, `iPhone`

**Comment by ncreated** [comment]
Hello @khsora34 👋. Thank you for providing example project, that is super handy, always 👌.

I've been playing with it for a while, but the problem doesn't seem to be coming from our SDK. I was able to make similar app setup with SPM and Carthage and it worked ("Product -> Export Localizations" ✅). Also, if trying with Cocoapods (in your sample project), it works from command-line:
```
xcodebuild -workspace LocalizationTest.xcworkspace -scheme LocalizationTest -exportLocalizations -localizationPath l8n

...

2023-04-24 11:10:06 Exporting LocalizationTest/Preview Content/Preview Assets.xcassets to en.xcloc...
2023-04-24 11:10:06 Exporting LocalizationTest/Assets.xcassets to en.xcloc...
2023-04-24 11:10:06 Exporting LocalizationTest/en.lproj/LocalizationTest-InfoPlist.strings to en.xliff...
2023-04-24 11:10:06 Writing
2023-04-24 11:10:06 An xcloc was created at /.../LocalizationTest/l8n/en.xcloc
```
The "Localization Export" option in Xcode doesn't seem to be much customizable, e.g. it can't be adjusted in scheme editor. On the other hand, the problem is caused by incorrect build order:

<table>
<tr>
<td>Product → Build</td>
<td>Product → Export Localization</td>
</tr>
<tr>
<tr>
<td>It builds: <code>PLCrashReporter</code> before <code>DatadogCrashReporting</code> 👍</td>
<td>It builds <code>DatadogCrashReporting</code> and no <code>PLCrashReporter</code> 👎, so it fails on <code>No such module 'CrashReporter'</code></td>
</tr>
<tr>
<td>
<img width="330" alt="ok-build" src="https://user-images.githubusercontent.com/2358722/233952831-d7e8e843-0d78-49e4-85bd-de0decebdb98.png">
</td>
<td>
<img width="499" alt="nok-l8n-export" src="https://user-images.githubusercontent.com/2358722/233952858-02055886-dbe3-4c77-b90d-1cd2bacefa69.png">
</td>
</tr>
</table>

I also tried disabling [LOCALIZATION_EXPORT_SUPPORTED](https://developer.apple.com/documentation/xcode/build-settings-reference) for our SDK, but it doesn't change anything (which makes sense, because the problem is in "build", not in "export").

Given that it works in `xcodebuild -exportLocalizations` and there is no problem with SPM or Carthage setups, I rather think this is likely an issue in Xcode (UI vs CLI) or Cocoapods. I couldn't find any opened report in neither of two. But with my best understanding, this is not a problem of `dd-sdk-ios`.

I know it doesn't help much (except CLI workaround), but let us know if we can do anything more to help you.

**Comment by khsora34** [comment]
Thank you so much for your answer, it was very insightful.

I tried however to follow the workaround, compiling with the CLI, but I still find the same problem with the sample project I sent. 

``` xcodebuild -workspace LocalizationTest.xcworkspace -scheme LocalizationTest -exportLocalizations -localizationPath l8n

...

~/Desktop/LocalizationTest/Pods/DatadogSDKCrashReporting/Sources/DatadogCrashReporting/PLCrashReporterIntegration/CrashReport.swift:8:8: error: no such module 'CrashReporter'
import CrashReporter
       ^

2023-04-24 10:13:30.333 xcodebuild[8992:80299] Writing error result bundle to /var/folders/lg/03vnyl_95nn6_qsq7l55vxmh0000gq/T/ResultBundle_2023-24-04_10-13-0030.xcresult
xcodebuild: error: Unable to build project for localization string extraction
	Reason: Failed to build.
	Please see the build logs for failure description.
```

Is it possible that there is a wrong dependency configured in the podspec? Maybe that leads to compiling `DatadogCrashReporting` before `PLCrashReporter`.

I did try with SPM and it seems to be working fine. However, I don't know if we'll be able to add SPM to our project.

Please let me know if you need more info from my setup that might be useful.

**Comment by ncreated** [comment]
@khsora34 👋.

> Is it possible that there is a wrong dependency configured in the podspec? Maybe that leads to compiling `DatadogCrashReporting` before `PLCrashReporter`.

If there would be any problem with podspecs, we will see it during regular build. Here, the problem is that "Product → Build" command works properly, but "Product → Export Localization" does not.

> I tried to follow the workaround, compiling with the CLI. However, I still find the same problem with the sample project I sent.

I did it another time (Xcode 14.3) and indeed it is faulty when trying in clean build, what I overlooked earlier - sorry. However, it seems that building incrementally can mitigate the problem:

1/ Just build (not I had to configure "Signing & Capabilities" in order to make it pass):
```
xcodebuild build -workspace LocalizationTest.xcworkspace -scheme LocalizationTest
```
2/ Then export localizations:
```
xcodebuild -workspace LocalizationTest.xcworkspace -scheme LocalizationTest -exportLocalizations -localizationPath l8n
```

What's funnier is that this works even from Xcode UI: first do "Product → Build" and then "Product → Export Localization" without cleaning the build folder in between. Also, that mitigation kind of makes sense - because of "Export Localization" being unable to build `CrashReporter` module, we leverage it being located from previous build command.

Except finding this workaround, I couldn't get anything new on this topic. I still suspect the problem is not in `dd-sdk-ios`. Let me know @khsora34 if you came up with anything new 🙏
